lottie/src/main/java/com/airbnb/lottie/animation/keyframe/DropShadowKeyframeAnimation.java

```diff
@@ -42,24 +46,27 @@ public DropShadowKeyframeAnimation(BaseKeyframeAnimation.AnimationListener liste
   }
 
   @Override public void onValueChanged() {
-    isDirty = true;
     listener.onValueChanged();
   }
 
-  public void applyTo(Paint paint) {
-    if (!isDirty) {
```

Comment:
@gpeal Would that look something like this?
```java
    // If the inputs haven't changed, we don't need to update the Paint
    if (direction.getValue() == lastAppliedDirection &&
        distance.getValue() == lastAppliedDistance &&
        radius.getValue() == lastAppliedRadius &&
        color.getValue() == lastAppliedColor &&
        opacity.getValue() == lastAppliedOpacity &&
        parentAlpha == lastAppliedParentAlpha) {
      return;
    }
```

(Assuming the `BaseKeyframeAnimation.getValue()` calls are cached and not expensive)

Reply:
Yup! The keyframe getValue() calls are designed to be pretty efficient but you can create local variables for them so that if you make it past the if, you can reuse the locals.